### DISCUSSION TITLE
How to rename private repository in GitHub for Desktop?

### DISCUSSION THREAD

**Discussion by IvanNextToJunior** [community/community#22147] [New to GitHub]
<p>Hello, everyone! Is that possible to rename a branch in private repository? If yes, how to do that? I have default security settings on that branch</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/ivannexttojunior">@IvanNextToJunior</a>, since you’re using GitHub Desktop, are you able to use the <strong>Rename…</strong> option under the <strong>Branch</strong> menu?</p>

**Comment by IvanNextToJunior** [comment]
<p>No, it’s grayed out</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/452f4219c11fac48668122f447aa7f6ba26d8c32.jpeg" data-download-href="/uploads/short-url/9S2c8gC6VYfp3T31oiEWy1HnDy2.jpeg?dl=1" title="Снимок экрана 2020-06-22 в 18.19.27" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/58988299/181069047-0d068a0b-9036-4f6d-9508-8b040f766017.jpeg" alt="Снимок экрана 2020-06-22 в 18.19.27" data-base62-sha1="9S2c8gC6VYfp3T31oiEWy1HnDy2" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/452f4219c11fac48668122f447aa7f6ba26d8c32_2_690x431.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/4/452f4219c11fac48668122f447aa7f6ba26d8c32_2_1035x646.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/452f4219c11fac48668122f447aa7f6ba26d8c32_2_1380x862.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/452f4219c11fac48668122f447aa7f6ba26d8c32_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Снимок экрана 2020-06-22 в 18.19.27</span><span class="informations">1440×900 265 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by steveward** [comment, marked as answer]
<p><a class="mention" href="/ivannexttojunior">@IvanNextToJunior</a> it looks like you are trying to rename your default branch, which currently is not possible in GitHub Desktop. The best solution is to follow these steps:</p>
<ol>
<li>Create a new branch off of <code>master</code>
</li>
<li>Publish the branch to your repository on GitHub</li>
<li><a href="https://help.github.com/en/github/administering-a-repository/setting-the-default-branch">Change your default branch on GitHub</a></li>
</ol>
<p>You’ll need to refresh your repository after you change the default branch – selecting the repository again in GitHub Desktop should refresh and grab the new default branch from the API. I hope that helps!</p>

**Comment by IvanNextToJunior** [comment]
<p>That’s actually the case. Thank you</p>
